TrackletAlgorithm/MatchCalculator.h

```diff
@@ -816,49 +816,49 @@ void MatchCalculator(BXType bx,
       switch (projseed) {
         case 0:
         if(FMMask<LAYER, PHISEC, TF::L1L2>()) {
-          fullmatch[0].write_mem(bx,bestmatch,nmcout1); // L1L2 seed
+          fullmatch[FMCount<LAYER, PHISEC, TF::L1L2>()].write_mem(bx,bestmatch,nmcout1); // L1L2 seed
           nmcout1++;
         }
         break;
         case 1:
         if(FMMask<LAYER, PHISEC, TF::L2L3>()) {
-          fullmatch[1].write_mem(bx,bestmatch,nmcout2); // L2L3 seed
+          fullmatch[FMCount<LAYER, PHISEC, TF::L2L3>()].write_mem(bx,bestmatch,nmcout2); // L2L3 seed
           nmcout2++;
         }
         break;
         case 2:
         if(FMMask<LAYER, PHISEC, TF::L3L4>()) {
-          fullmatch[2].write_mem(bx,bestmatch,nmcout3); // L3L4 seed
+          fullmatch[FMCount<LAYER, PHISEC, TF::L3L4>()].write_mem(bx,bestmatch,nmcout3); // L3L4 seed
           nmcout3++;
         }
         break;
         case 3:
         if(FMMask<LAYER, PHISEC, TF::L5L6>()) {
-          fullmatch[3].write_mem(bx,bestmatch,nmcout4); // L5L6 seed
+          fullmatch[FMCount<LAYER, PHISEC, TF::L5L6>()].write_mem(bx,bestmatch,nmcout4); // L5L6 seed
```

Comment:
This is a some what complicated solution I came up with to count how many slots are not masked. That way, seeding in L5L6 (which would be slot 3) is placed in the correct slot (slot 2 if the TBHelper is only reading L1L2 and L5L6). This relies on the mask to agree with the number of files TBHelper loads, which seems reasonable.

Reply:
@bryates I'm not sure I understand what's happening here. I thought the `FMMask` essentially told you which FM memories are "on" for a given MC. That way, the size of the `fullmatch` array can be the same for every MC, but HLS will only generate ports for the memories that are enabled by the mask.

So why would the index of, e.g., the L5L6 FM memory ever change?

Reply:
`FMMask` disables ports so Vivado will trim them, but this does not change the index, which was hard coded to 3 in the past. `FMCount` allows it to determine how many ports are actually active, and allocate the correct spot. I needed this because the `TBHelper` uses vectors that are only as large as the active ports.